cpp/benchmarks/io/json/json_reader_option.cpp

```diff
@@ -0,0 +1,124 @@
+      .lines(true)
+      .na_rep("null")
+      .rows_per_chunk(100'000);
+  cudf::io::write_json(write_opts);
+
+  cudf::io::json_reader_options read_options =
+    cudf::io::json_reader_options::builder(source_sink.make_source_info())
+      .lines(true)
+      .normalize_single_quotes(normalize_single_quotes_bool)
+      .mixed_types_as_string(mixed_types_as_string_bool)
+      .recovery_mode(recovery_mode_enum);
+
+  size_t const chunk_size = cudf::util::div_rounding_up_safe(source_sink.size(), num_chunks);
+  auto mem_stats_logger   = cudf::memory_stats_logger();
+  state.set_cuda_stream(nvbench::make_cuda_stream_view(cudf::get_default_stream().value()));
+  state.exec(
+    nvbench::exec_tag::sync | nvbench::exec_tag::timer, [&](nvbench::launch& launch, auto& timer) {
+      try_drop_l3_cache();
+      cudf::size_type num_rows_read = 0;
+      uint64_t num_cols_read        = 0;
+      timer.start();
+      for (uint64_t chunk = 0; chunk < num_chunks; chunk++) {
+        switch (RowSelection) {
+          case row_selection::ALL: break;
+          case row_selection::BYTE_RANGE:
+            read_options.set_byte_range_offset(chunk * chunk_size);
+            read_options.set_byte_range_size(chunk_size);
+            break;
+          default: CUDF_FAIL("Unsupported row selection method");
+        }
+
+        auto const result = cudf::io::read_json(read_options);
+
+        num_rows_read += result.tbl->num_rows();
+        num_cols_read = result.tbl->num_columns();
+      }
+
+      timer.stop();
+      CUDF_EXPECTS(num_rows_read == view.num_rows(), "Benchmark did not read the entire table");
+      CUDF_EXPECTS(num_cols_read == num_cols, "Unexpected number of columns");
+    });
+
+  auto const elapsed_time   = state.get_summary("nv/cold/time/gpu/mean").get_float64("value");
+  auto const data_processed = data_size * num_cols / view.num_columns();
+  state.add_element_count(static_cast<double>(data_processed) / elapsed_time, "bytes_per_second");
+  state.add_buffer_size(
+    mem_stats_logger.peak_memory_usage(), "peak_memory_usage", "peak_memory_usage");
+  state.add_buffer_size(source_sink.size(), "encoded_file_size", "encoded_file_size");
+}
+
+NVBENCH_BENCH_TYPES(
+  BM_json_read_options,
+  NVBENCH_TYPE_AXES(
+    nvbench::enum_type_list<row_selection::ALL, row_selection::BYTE_RANGE>,
+    nvbench::enum_type_list<normalize_single_quotes::NO, normalize_single_quotes::YES>,
+    nvbench::enum_type_list<mixed_types_as_string::NO, mixed_types_as_string::YES>,
+    nvbench::enum_type_list<recovery_mode::RECOVER_WITH_NULL, recovery_mode::FAIL>))
+  .set_name("json_read_all_lines")
+  .set_type_axes_names(
+    {"row_selection", "normalize_single_quotes", "mixed_types_as_string", "recovery_mode"})
+  .set_min_samples(6)
+  .add_int64_axis("num_chunks", {1})
+  .add_int64_axis("num_cols", {64})
+  .add_int64_power_of_two_axis("table_size", nvbench::range(26, 29, 1));
```

Comment:
is this temporary (to make those lovely charts), or planned to be merged?

Reply:
I think the `table_size` or `num_cols` axes need not be merged, but I think we can retain the `num_chunks` (with some nvbench range) to showcase the byte range performance.